src/backends/pyplot.jl

```diff
@@ -1320,7 +1315,7 @@ function py_add_legend(plt::Plot, sp::Subplot, ax)
                         color = py_color(single_color(get_linecolor(series, clims)), get_linealpha(series)),
                         linewidth = py_thickness_scale(plt, clamp(get_linewidth(series), 0, 5)),
                         linestyle = py_linestyle(:path, get_linestyle(series)),
-                        marker = py_marker(_cycle(series[:markershape], 1)),
+                        marker = py_marker(_cycle(series[:markershape], 1)),  # Need to adjust markersize to match the legend box size
```

Comment:
Here, we can specify `markersize` too, but it only needs to trigger when `thickness_scaling` is used.